c++/src/kj/async.h

```diff
@@ -388,6 +393,12 @@ class ForkedPromise {
   Promise<T> addBranch();
   // Add a new branch to the fork.  The branch is equivalent to the original promise.
 
+  _::BranchForCoAwaitPromise<T> addBranchForCoAwait();
+  // Add a new branch to the fork. 
+  // Like addBranch(), but use this version when you will immediately `co_await` the result.
+  // This version avoids allocation by leveraging the coroutine frame. You must not store the
+  // result, the only thing you may do with it is immediately `co_await` it.
```

Comment:
Isn't the requirement that we must not _move_ the result, rather than not store it? If so, could we make it `KJ_DISALLOW_COPY_AND_MOVE` and rely on guaranteed copy elision to make sure wherever it gets stored -- either as a temporary in `co_await fork.addBranch...()` or as an lvalue in `auto branch = fork.addBranch...()` -- never changes addresses?